tests/jballerina-unit-test/src/test/resources/test-src/reachability-analysis/unreachability_test.bal

```diff
@@ -932,3 +932,53 @@ function testUnreachableCodeWithTypeNarrowing8() {
         never _ = a; // unreachable code
     }
 }
+
+public function testUnreachableCodeWithTypeNarrowing9() {
+    int? a = 10;
+    string b = "A";
+    if b == "" || a == 0 {
+        int? _ = a;
+    } else if a is int || a is () {
+        int? _ = a;
+    } else if a is () {
+        never _ = a; // unreachable code
```

Comment:
Once we introduce the change to allow `never _ = a;` in narrowed contexts, this will no longer be unreachable, right?
 Shall we change the tests to use a non-never type instead?

Reply:
We can do this later anyway.